lineapy/graph_reader/graph_printer.py

```diff
@@ -67,6 +68,14 @@ def get_node_type_count(self, node_type: NodeType) -> int:
     def get_node_type_name(self, node_type: NodeType) -> str:
         return f"{pretty_print_node_type(node_type)}_{self.get_node_type_count(node_type)}"
 
+    def get_node_type_name_from_node(self, node: Node) -> str:
```

Comment:
suggest a different name. using get_node_type_name as the prefix suggests that its the same thing as that other function except diff arguments. the behavior here seems to be "caching" the results of the other function so a better name would be get_cached_node_type_name